docs/Developers/Profile_Bears.rst

```diff
@@ -0,0 +1,74 @@
+Profile Bears
+=============
+
+This document provides an overview of coala's Profiling Interface. The
+Profiling Interface provides abilities to profile Bear code to optimize its
+performance.
+
+The profiler will start by profiling the run() method of bears because this is
+the part where bear writers will spend time on, as rest of the part like loading
+the files, collecting the settings, etc. are done by coala itself.
+
+.. note::
+
+    Enabling both profiler and debugger on a bear at the same time will result
+    in error.
+
+Bear writers will have the ability to directly dump the raw profile output
+either on current working directory or to a specified directory name, which can
+be further used for examination of profiler stats with the help of different
+modules like ``pstats`` or ``snakeviz``.
+
+Usage
+-----
+
+coala's Profiler accepts an additional parameter, directory name or directory
+path where profiled files will be dumped, if ``True`` is provided profiled
+files will be saved to current working directory. If specified directory does
+not exists it will create one and if path specified is already a file path then
```

Comment:
If *the* specified directory does not exist~~s~~ it will be created ~~create one~~. If the specified path points to an already existing file a `FileExistsError` is thrown.